### PR TITLE
reorg solution file

### PR DESCRIPTION
**Pull request by dsyme**
This renames

    .\vsintegration\src\Deployment.sln --> .\VisualFSharp.sln
    Salsa.dll --> VisualFSharp.Salsa.dll
    Unittests.dll --> VisualFSharp.Unittests.dll

It also 
* adds a couple of virtual folders to VisualFSharp.dln (was called Deployment.sln)
* removes a couple of test files that weren't being referenced anywhere

The aim is to make it easier to contribute to the Visual F# Tools by improving the organization of things. The visual appearance of VisualFSharp.sln is shown below

capture

### CODE REVIEW

**Comment by enricosada** [comment]
maybe we can also show a clean cut between FSharp.Core (a library) and the compiler,

**Comment by dsyme** [comment]
OK done, please see below

capture

**Comment by rodrigovidal** [comment]
Wow this way looks so much better! :+1:

**Comment by enricosada** [comment]
Fine for me

**Comment by dsyme** [comment]
Pulling after discussing with @KevinRansom.  Microsoft still has an internal build process used to package the Visual F# Tools which is separate to the process used here, and this PR might affect that because the test DLLs are being renamed.  But we think that the test DLLs no longer need to be built as part of that. 

In any case a big thanks to @KevinRansom in advance for dealing with any fallout of this w.r.t. the internal Microsoft build process (a pain point we hope will go away by moving to purely external build process eventually)

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Lazy.fs [hunk 1]
```diff
@@ -1,172 +0,0 @@
-// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Lazy.fs]: @dsyme why this file is deleted?

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigIntStruct.fs [hunk 2]
```diff
@@ -1,374 +0,0 @@
-// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigIntStruct.fs]: that too, why is deleted? it's duplicated?

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigNum.fs [hunk 3]
```diff
@@ -1,320 +0,0 @@
-// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigNum.fs]: another file deleted

**Comment by enricosada** [comment]
@dsyme some test file are deleted, it's intentional (duplicated) or an error?

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigNum.fs (hunk 3 above)
Comment (dsyme) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigNum.fs]: BigNum is the BigRational from the power pack, removed a long time ago

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigIntStruct.fs (hunk 2 above)
Comment (dsyme) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Math/BigIntStruct.fs]: This file was not included in any active testing. BigInteger is now part of the .NET Framework and tested there.  Also we have some tests in BigIntType.fs

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Lazy.fs (hunk 1 above)
Comment (dsyme) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Lazy.fs]: This file was not included in any active testing. It's duplicated in LazyType.fs, and in any case the type is part of the .NET Fx and tested there

**Comment by dsyme** [comment]
Yes, intentional, see comments above, thanks for checking!

**Comment by dsyme** [comment]
I just wanted to mention that I found the VisualFSharp.sln file _much_ easier to work with this evening - I could actually find the necessary code and the appropriate unit tests! :)

**Pull request merged**